### PR TITLE
initial commit of kma_index

### LINKED ISSUE #7235: new module: kma index
I plan to add a new nf-core module for [kma index](https://bitbucket.org/genomicepidemiology/kma/src/master/).

Checklist:

- [x] kma index code
- [x] kma index stub code
- [x] kma index test
- [x] kma index stub test

### PR DESCRIPTION
**Pull request by Krannich479**
## PR checklist

Closes #7235 

- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`

### CODE REVIEW

File: modules/nf-core/kma/index/main.nf [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/kma:1.4.15--he4a0461_0' :
+        'biocontainers/kma:1.4.15--he4a0461_0' }"
+
+    input:
+    tuple val(meta), path(fasta)
+
+    output:
+    tuple val(meta), path("${meta.id}.kmaindex.*"),   emit: db
```
Comment (famosab) [inline modules/nf-core/kma/index/main.nf:14]: Please make the outputname flexible :)
```suggestion
    tuple val(meta), path("*.kmaindex.*"),   emit: db
```
even better if a custom name would be possible, but sometimes this is difficult with db

File: modules/nf-core/kma/index/main.nf [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    END_VERSIONS
+    """
+
+    stub:
+    def prefix  = task.ext.prefix ?: "${meta.id}.kmaindex"
+    """
+    touch ${prefix}.comp.b
+    touch ${prefix}.length.b
+    touch ${prefix}.name
+    touch ${prefix}.seq.b
```
Comment (famosab) [inline modules/nf-core/kma/index/main.nf:42]: here i do not understand why single files are created - this seems not to fit just one output (the db)

File: modules/nf-core/kma/index/meta.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  - db:
+    - meta:
+        type: map
+        description: |
+          Groovy Map containing sample information
+          e.g. `[ id:'reference' ]`
+    - '${meta.id}.kmaindex.*':
+        type: file
+        description: KMA index files
+        pattern: "*.{comp.b,length.b,name,seq.b}"
```
Comment (famosab) [inline modules/nf-core/kma/index/meta.yml:42]: if all of them are created then declare each of those as output file :)

**Review by famosab — COMMENTED** [review]
Looks good already :)

**Comment by SPPearce** [comment]
Yes, I think that is the standard method to deal with index generation. The individual files on their own are useless, they are required in combination.

**Comment by Krannich479** [comment]
kmaIndex_gitpod

Docker test succeeds in gitpod.

**Comment by Krannich479** [comment]
@jfy133 is this the kind of proof you proposed?

**Review by jfy133 — APPROVED** [review]

**Pull request merged**